src/Microsoft.AspNet.Routing/Template/RouteTemplate.cs

```diff
@@ -41,6 +41,16 @@ public RouteTemplate(List<TemplateSegment> segments)
 
         public IList<TemplateSegment> Segments { get; private set; }
 
+        public TemplateSegment GetSegment(int index)
+        {
+            if (index < 0)
+            {
+                throw new IndexOutOfRangeException();
```

Comment:
Sure. Practically speaking though this will never happen. This is mostly here as documentation for the fact that I don't expect it to happen.